code/__DEFINES/dcs/signals.dm

```diff
@@ -1414,3 +1414,7 @@
 
 /// Called on the organ when it is removed from someone (mob/living/carbon/old_owner)
 #define COMSIG_ORGAN_REMOVED "comsig_organ_removed"
+
+// Caltrop Signals
+/// Called when a caltrop with the CALTROP_UPDATE_DAMAGE flag gets its damage updated. For example, (/obj/effect/decal/cleanable/ants)
+#define COMSIG_CALTROP_UPDATE "comsig_caltrop_update"
```

Comment:
this signal is shit. it's a proc disguised as a signal.

Reply:
How else do I send a proc to a component?

Reply:
you shouldn't.  https://tgstation13.org/phpBB/viewtopic.php?f=5&t=22674

Reply:
Reading through the [new location where the guide is](https://hackmd.io/@tgstation/SignalsComponentsElements) according to the original author of it, it really just points towards signals being the way to interact with a component or element, as far as I can tell.
I can't really think of any other way to transfer a change in a variable in one object which can be manipulated by a player into a component other than a signal, but I am always one to admit my rather amateur coding knowledge, so if there is another way to do this then I'm not privy to it yet.

Reply:
we can discuss this more in-depth in discord if you'd like. that might be easier

Reply:
bad usage of signals. Use the COMPONENT_DUPE_UNIQUE_PASSARGS on components and then just simply add the component again with the new damage value.
Then you can override /datum/component/InheritComponent to update your component.

Reply:
Oh yea looking at it, this is exactly what I needed. I completely forgot that components had `dupe_mode` as a variable, thanks for reminding me of this.